src/codegen.cpp

```diff
@@ -3994,6 +4016,11 @@ static jl_cgval_t emit_call(jl_codectx_t &ctx, jl_expr_t *ex, jl_value_t *rt)
         JL_I::intrinsic fi = (intrinsic)*(uint32_t*)jl_data_ptr(f.constant);
         return emit_intrinsic(ctx, fi, args, nargs - 1);
     }
+    else if (f.constant && f.constant == jl_builtin_modifyfield) {
+        jl_cgval_t ret(ctx.builder.getContext());
+        if (emit_call_modify(ctx, &ret, f, ex, rt))
+            return ret;
+    }
```

Comment:
Why is this not getting handled by `emit_builtin_call` / `emit_f_opfield` like normal? It doesn't seem like it should be requiring special call semantics.